File: src/main/java/com/mutableconst/sql/NamedPreparedQuery.java
package com.mutableconst.sql;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

import com.mutableconst.sql.util.SqlResult;

public class NamedPreparedQuery extends NamedPreparedStatement {

    public NamedPreparedQuery(String preparedSql) {
        super(preparedSql);
    }

    @Override
    public SqlResult execute(Connection connection) throws SQLException {
        PreparedStatement query = createPreparedStatement(connection);
        return new SqlResult(true, query.executeQuery());
    }
}

File: src/main/kotlin/com/mutableconst/sql/NamedPreparedQuery.kt
package com.mutableconst.sql

import com.mutableconst.sql.util.SqlResult
import java.sql.Connection
import java.sql.SQLException

class NamedPreparedQuery(preparedSql: String) : NamedPreparedStatement(preparedSql) {

    @Throws(SQLException::class)
    override fun execute(connection: Connection, parameters: Map<String, Any>): SqlResult {
        val query = createPreparedStatement(connection, parameters)
        return SqlResult(true, query.executeQuery(), query)
    }
}
